user_tools/src/spark_rapids_tools/tools/qualx/qualx_main.py

```diff
@@ -52,26 +53,44 @@
 )
 from spark_rapids_pytools.common.utilities import Utils
 from tabulate import tabulate
-from xgboost.core import XGBoostError
+from xgboost.core import XGBoostError, Booster
 
 logger = get_logger(__name__)
 
 
-def _get_model(platform: str, model: Optional[str]):
-    if not model:
+def _get_model(platform: str,
+               model: Optional[str] = None,
+               custom_model_file: Optional[str] = None) -> Booster:
```

Comment:
Not sure we need another argument for `custom_model_file` here.  Basically, the custom model path should just be passed to the `model` arg.

The original `model` arg had the following behavior:
1. if None, try to use a pre-trained `platform` model, e.g. `onprem`.
2. if specified, try to locate the model from the pre-trained model directory, e.g. `combined`.
3. if specified and not found in pre-trained model directory, try to locate the model from the path as-is.  If not found, this would raise an error.

So, the custom model path should fall into case 3.

Also, I'm not aware of how `CspPath` works, but I assume it would download from remote to local disk, s.t. `xgb_model.load_model()` works as usual?

Reply:
> Basically, the custom model path should just be passed to the model arg.

Thanks @parthosa and @leewyang 

I fixed anyway but this is my feedback about this piece of:

- In order to use "`model`" as a string name, the file has to be saved inside resources which is deprecated by having an input file that can be loaded from anywhere.

- The name `custom_file` makes the code more readable because anywaone can understand that this is "meant" to be a file. On the other hand, the word `model`  is overused all over the place which makes it very hard to read the code.

> Also, I'm not aware of how CspPath works, but I assume it would download from remote to local disk, s.t. xgb_model.load_model() works as usual?

`CspPath` is our internal implementation to handle different FileSystems.
It has good support to do some validations and checks.

- We have infrastructure to copy the files from remote filesystem to local ones. I haven't add that check in that code here. I put a `TODO` to revisit it later.
- The new code checks whether "model" is a valid file or not. then it will parse it as a file.

Reply:
OK, thanks for the explanation and updates!